source/common/stats/symbol_table_impl.h

```diff
@@ -242,11 +279,20 @@ class SymbolTable {
  * Holds backing storage for a StatName. Usage of this is not required, as some
  * applications may want to hold multiple StatName objects in one contiguous
  * uint8_t array, or embed the characters directly in another structure.
+ *
+ * This is intended for embedding in other data structures that have access
+ * to a SymbolTable. StatNameStorage::free(symbol_table) must be called prior
+ * to allowing the StatNameStorage object to be destructed, otherwise an assert
+ * will fire to guard against symbol-table leaks.
+ *
+ * Thus this class is inconvenient to directly use as temp storage for building
+ * a StatName from a string. Instead it should be used via StatNameTempStorage.
  */
 class StatNameStorage {
 public:
   StatNameStorage(absl::string_view name, SymbolTable& table);
```

Comment:
With three constructors, it might be nice to have a comment inline about the types of callsites which might use each of them. Otherwise it might be easy to come along later and decide to "clean this up" by consolidating the constructors.